test/extensions/tracers/xray/xray_tracer_impl_test.cc

```diff
@@ -72,12 +72,39 @@ TEST_F(XRayDriverTest, XRayTraceHeaderSamplingUnknown) {
   auto span = driver.startSpan(tracing_config_, request_headers_, operation_name_, start_time,
                                tracing_decision);
   // sampling should fall back to the default manifest since:
-  // a) there is sampling decision in the X-Ray header
+  // a) there is no valid sampling decision in the X-Ray header
```

Comment:
Shouldn't there be some update to this test case? I'm confused as I don't see any behavior change here and the new test case below doesn't look like it actually tests the broken case?

Reply:
The broken case was when `XRayTraceHeader` doesn't contain `"Sampled="`. This wasn't covered under unit testing before so I am adding this test in L82 below where `XRayTraceHeader=Root=1-272793;Parent=5398ad8;`

Earlier when we wrote the unit test we didn't consider a case where `"Sampled="` will not be included in the trace header. Apparently [ELB](https://docs.aws.amazon.com/xray/latest/devguide/xray-services-elb.html) can send a trace header without making `Sampled=1` or `Sampled=0` decision and let's the subsequent service decide whether to sample or not.


Now coming to this question, this test doesn't need any update as `XRayTraceHeader=Root=1-272793;Parent=5398ad8;Sampled=` can also be valid scenario and default sampling rule (defalt_target=1) will sample this case as usual.

If we notice in Line85 below I am overriding a default sampling_rules with a custom sampling_rules.

Reply:
OK sounds good thanks for the clarification.